twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillRunnerService.java

```diff
@@ -157,11 +158,25 @@ public YarnTwillRunnerService(YarnConfiguration config, String zkConnect) {
    * @param locationFactory Factory to create {@link Location} instances that are readable and writable by this service
    */
   public YarnTwillRunnerService(YarnConfiguration config, String zkConnect, LocationFactory locationFactory) {
+    this(config, zkConnect, locationFactory, true);
+  }
+
+  /**
+   * Creates an instance.
+   *
+   * @param config Configuration of the yarn cluster
+   * @param zkConnect ZooKeeper connection string
+   * @param locationFactory Factory to create {@link Location} instances that are readable and writable by this service
+   * @param enableSecureStoreUpdate Whether to update SecureStore periodically
+   */
+  public YarnTwillRunnerService(YarnConfiguration config, String zkConnect, LocationFactory locationFactory,
+                                boolean enableSecureStoreUpdate) {
     this.yarnConfig = config;
     this.yarnAppClient = new VersionDetectYarnAppClientFactory().create(config);
     this.locationFactory = locationFactory;
     this.zkClientService = getZKClientService(zkConnect);
     this.controllers = HashBasedTable.create();
+    this.enableSecureStoreUpdate = enableSecureStoreUpdate;
```

Comment:
Better use a configuration passed in from the `YarnConfiguration`. The keys should be defined in `Configs.Keys` class.